gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiServiceImpl.java

```diff
@@ -2004,75 +2020,85 @@ public void delete(ExecutionContext executionContext, String apiId) {
 
             Api api = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
 
-            if (api.getLifecycleState() == LifecycleState.STARTED) {
-                throw new ApiRunningStateException(apiId);
+            if (DefinitionContext.isKubernetes(api.getOrigin())) {
```

Comment:
not part of this issue but i really think we can improve this service to avoid this kind of `if`. and refactor all this service